src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -939,21 +943,19 @@ public RequestLineStatus TakeStartLine(SocketInput input)
 
                     if (httpVersion == null)
                     {
-                        RejectRequest(RequestRejectionReason.MissingHTTPVersion);
+                        RejectRequest(RequestRejectionReason.InvalidRequestLine, start.GetAsciiStringEscaped(end, MaxInvalidRequestLineChars));
                     }
                     else if (httpVersion != "HTTP/1.0" && httpVersion != "HTTP/1.1")
                     {
-                        RejectRequest(RequestRejectionReason.UnrecognizedHTTPVersion);
+                        RejectRequest(RequestRejectionReason.UnrecognizedHTTPVersion, httpVersion);
```

Comment:
Couldn't httpVersion be arbitrarily long? The above comment about it being possible for a valid version to span more than 2 blocks is also wrong.